### PR TITLE
refactor(turborepo): linting pipeline

### PR DESCRIPTION
**Pull request by Leomotors**
Configured eslint to make it best for turborepo

My suggested approach:
All package use `.eslintrc` at root level and if they want to override they can add their own `.eslintrc`

**Need comment**: Is this style of extending rules okay?

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,5 +0,0 @@
-# Next.js
```
Comment (saenyakorn) [inline .eslintignore:1]: why delete this na? I think we don't need to lint artifact or cache files

File: apps/reg-scraper/package.json [hunk 2]
```diff
@@ -44,4 +43,4 @@
     "tough-cookie": "^4.1.2",
     "winston": "^3.8.2"
   }
-}
+}
```
Comment (saenyakorn) [inline apps/reg-scraper/package.json:46]: not sure why EOF is missing every `package.json` lol

File: packages/config/package.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+  "name": "@cgr/config",
```
Comment (saenyakorn) [inline packages/config/package.json:2]: not sure about the naming. `config` is too wide. Do we have other config than Eslint?

**Comment by saenyakorn** [comment]
Do not merge this before someone fix admin-web build failed

**Comment by saenyakorn** [comment]
Also, could you enable `lint` in `reusable-ci.yaml`. Thank you

File: .eslintignore (hunk 1 above)
Comment (Leomotors) [reply .eslintignore:1]: Moved to `ignorePatterns` in `.eslintrc`

File: apps/reg-scraper/package.json (hunk 2 above)
Comment (Leomotors) [reply apps/reg-scraper/package.json:46]: My vscode config 💀, we should get some husky

File: .prettierrc.js [hunk 4]
```diff
@@ -1,7 +1,12 @@
-module.exports = {
+// @ts-check
+
+/** @type {import('prettier').Config} */
+const config = {
   singleQuote: true,
   semi: false,
   printWidth: 100,
   tabWidth: 2,
   trailingComma: 'es5',
```
Comment (saenyakorn) [inline .prettierrc.js:9]: Please apply `sort-import` here. The order is

1. `@nestjs/**`
2. `react/**`
3. 3rd Party
4. 
5. `@/**` (alias absolute path for the app)
6. `@cgr/**` (for local packages)
7. ./ 
8. ../

Don't for get to separate the imports using `newline`

Or, you can create `prettier-plugin-base`, `prettier-plugin-react` and `prettier-plugin-nest` in `@cgr/project-config` to configure different import order for different based framework. 

PS. We already installed the plugin in the root package.json. But, somehow, we forgot to use the plugin. lol

Comment (bombnp) [reply .prettierrc.js:9]: plugin is `@trivago/prettier-plugin-sort-imports`

File: packages/project-config/eslint-base.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  parser: '@typescript-eslint/parser',
+  plugins: ['@typescript-eslint/eslint-plugin'],
+  extends: [
+    'eslint:recommended',
+    'plugin:@typescript-eslint/recommended',
+    'plugin:prettier/recommended',
+  ],
+  rules: {
+    '@typescript-eslint/no-empty-function': 'off',
+  },
```
Comment (bombnp) [inline packages/project-config/eslint-base.js:15]: I want u to use `eslint-plugin-unused-imports` as well. Good shit. It automatically removes imports when used with `eslint --fix` in vscode settings.

**Comment by bombnp** [comment]
so are we gonna move tsconfigs to `@cgr/project-config` as well? I think we should.

**Comment by bombnp** [comment]
also i fixed the build problem with #529, rebase your branch to see if CI is fixed

**Comment by Leomotors** [comment]
To keep PR small, let's go with this for now and then I can work on `prettier` on next PR

**Review by saenyakorn — APPROVED** [review]
For me, LGTM

**Review by bombnp — APPROVED** [review]

**Pull request merged**